### PR TITLE
Allow for downloading of the certs, keys

### PR DESCRIPTION
**Pull request by d1str0**
All require auth

GET /api/certificate/{id}/cert
Returns the certificate file

GET /api/certificate/{id}/issuer
Returns the issuer's certificate as a file

GET /api/certificate/{id}/privkey
Returns the certificate's private key as a file

### CODE REVIEW

**Comment by d1str0** [comment]
Each of the above endpoints now requires Authorization: Secret {secret} instead of the usual Bearer and JWT. This means certs can be downloaded ONLY if you know the secret per certificate.

Certificate secrets generated randomly in the same manner as Super Admin passwords.

File: auth/utils.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+// randBytes is the number of bytes of entropy for SA password
+const randBytes = 16
+
+// NewPassword generates cryptographically secure random bytes, base64
+// encodes it, and returns it.
+func NewPassword() string {
+	c := randBytes
+	b := make([]byte, c)
```
Comment (phildebus) [inline auth/utils.go:16]: Why use c at all? why not just randBytes

**Review by phildebus — APPROVED** [review]

**Review by brooks42 — APPROVED** [review]
LGTM

File: auth/utils.go (hunk 1 above)
Comment (d1str0) [reply auth/utils.go:16]: no raisin. It was some copypasta.

**Pull request merged**